explainer.md

```diff
@@ -0,0 +1,447 @@
+
+* Root Transitions : The full page content animates between 2 web pages with an optional static UI element on top. [Shared axis](https://material.io/design/motion/the-motion-system.html#shared-axis) shows an example.
+* Shared Element Transitions : A persistent UI element morphs into another (which could be a UI element on the next page or the whole page) changing its shape and content. [Container transform](https://material.io/design/motion/the-motion-system.html#container-transform) shows an example.
+* Entry/Exit Transitions : A UI element animates as it exits or enters the screen. This [issue](https://github.com/WICG/shared-element-transitions/issues/37) shows an example.
+
+These transitions should be feasible in SPAs (Single Page Apps) and MPAs (Multi Page Apps).
+
+# Design
+Let's take the example below which shows how the API can be used by a developer to animate the background and a shared element on a same origin navigation (MPA). The SPA equivalent of this case is one where the old document is mutated into the new document via DOM APIs. See [API Extensions](#api-extensions) for more code examples.
+
+### Old Document
+```
+<html>
+<head>
+  <style>
+    body {
+      background-color: blue;
+    }
+    .animated {
+      position: relative;
+      top: 300px;
+      width: 100px;
+      height: 100px;
+      background-color: red;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" sharedId="header-id" id="header">Shared Element</div>
+</body>
+</html>
+```
+
+### New Document
+```
+<html>
+<head>
+  <style>
+    body {
+      background-color: lightblue;
+    }
+    .animated {
+      position: relative;
+      top: 100px;
+      width: 200px;
+      height: 200px;
+      background-color: green;
+    }
+    
+    ::shared-container(#header-id) {
+      animation: ::shared-container-header-id 1s ease-in;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" sharedId="header-id" id="header">Shared Element</div>
+</body>
+</html>
+```
+
+The steps taken by the browser during the transition are as follows.
+
+1. When a navigation is initiated on the old Document, create the following pseudo elements in the top layer[^1]. Note that a shared element must not be nested inside another shared element :
+
+    a. A container and child replaced element for each element with the sharedId attribute. These are identified via ::shared-container(#sharedId) and ::shared-old(#sharedId) respectively.
+
+    b. A replaced element for the root/html element identified via ::shared-old(#root).
+    
+    c. The box hierarchy in the top layer stacking context is :
+    
+    ```
+    ├───shared-old(root)
+    ├───shared-container(#header)
+         ├───shared-old(#header)
+    ```
+
+2. Apply the following UA stylesheet to the pseudo elements on the old page :
+```
+// "root" is a reserved keyword for the html element.
+::shared-old(#root), ::shared-container(#header-id) {
```

Comment:
We won't be able to get away with reserved keywords here. I suggest renaming to ::shared-old-root.

Reply:
Done.